unicsv.cc

```diff
@@ -1010,7 +1048,7 @@ unicsv_parse_one_line(const QString& ibuf)
       switch (unicsv_fields_tab[column]) {
 
       case fld_gc_id:
-        gc_data->id = value.toInt();
+        gc_data->id = value.toLongLong();
```

Comment:
Would it be more clear if  we used the ok parameter of toLongLong instead of relying on the result being zero when conversion fails (which is guaranteed by toLongLong).
Also a comment here saying if conversion succeeded then we assume the input is a geocache ID, and otherwise we assume it is a geocache reference code.

Reply:
@tsteven4  Might also make sense to explicitly add a base, otherwise this would accept hex and octals with 0x and 0 prefix.

> It disturbs me when we do an echo test that doesn't echo what we read

Yes indeed, but there doesn't seem to be the corresponding output code (besides a partial implementation in deprecated/geoniche.cc).

> In your new test we are reading reference codes, not geocache ids

The terminology was not entirely clear to me while implementing this, that might show in some places. An unrepresentative search engine query suggest that GC-Code and GC-ID are mostly used. The term "reference code" may be a generalization that also includes trackable codes starting with "TB". But I'm not sure.